invenio_communities/communities/records/systemfields/access.py

```diff
@@ -8,19 +8,54 @@
 
 """Community access system field."""
 
+from enum import Enum
+
 from invenio_records.systemfields import SystemField
 
 
+class EnumMixin:
+    """Mixin for enum functionalities."""
+
+    @classmethod
+    def validate(cls, level):
+        return level in cls._member_names_
+
+
+class VisibilityEnum(Enum, EnumMixin):
+    """Enum defining access visibility."""
+
+    PUBLIC = "public"
+
+    RESTRICTED = "restricted"
+
+
+class MemberPolicyEnum(Enum, EnumMixin):
+    """Enum defining member policies."""
+
+    OPEN = "open"
+
+    CLOSED = "closed"
+
+
+class RecordPolicyEnum(Enum, EnumMixin):
+    """Enum defining record policies."""
+
+    OPEN = "open"
+
+    CLOSED = "closed"
+
+
+class ReviewPolicyEnum(Enum, EnumMixin):
+    """Enum defining review policies."""
+
+    OPEN = "open"
+
+    CLOSED = "closed"
+
+
 class CommunityAccess:
     """Access management per community."""
 
-    # important: the order in tuple matters
```

Comment:
@kpsherva I could not find a usage where the order affects the logic, what was the use case? (it would be easily fixable by overwritting `__iter__`, I just wanted to make sure before implementing it.